python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -93,6 +110,108 @@ def serialize(self) -> str:
         return json.dumps(storage_dict)
 
 
+def execute_asset_backfill_iteration(
+    backfill: "PartitionBackfill", workspace: BaseWorkspaceRequestContext, instance: DagsterInstance
+) -> None:
+    """
+    Runs an iteration of the backfill, including submitting runs and updating the backfill object
+    in the DB.
+    """
+    from dagster._core.execution.backfill import BulkActionStatus
+
+    asset_graph = ExternalAssetGraph.from_workspace_request_context(workspace)
+    if backfill.serialized_asset_backfill_data is None:
+        check.failed("Asset backfill missing serialized_asset_backfill_data")
+
+    asset_backfill_data = AssetBackfillData.from_serialized(
+        backfill.serialized_asset_backfill_data, asset_graph
+    )
+
+    result = execute_asset_backfill_iteration_inner(
+        backfill_id=backfill.backfill_id,
+        asset_backfill_data=asset_backfill_data,
+        instance=instance,
+        asset_graph=asset_graph,
+    )
+
+    updated_backfill = backfill.with_asset_backfill_data(result.backfill_data)
+    if result.backfill_data.is_complete():
+        updated_backfill = updated_backfill.with_status(BulkActionStatus.COMPLETED)
+
+    for run_request in result.run_requests:
```

Comment:
you should probably be yielding `None` in loops, to give the daemon to heartbeat.  o.w. this can cause the daemon to get killed and restarted in a crash loop.

Reply:
👍